core/src/main/java/com/opensymphony/xwork2/config/providers/XmlConfigurationProvider.java

```diff
@@ -874,13 +874,16 @@ protected String guessResultType(String type) {
         if (allowedMethodsEls.getLength() > 0) {
             // user defined 'allowed-methods' so used them whatever Strict DMI was enabled or not
             allowedMethods = new HashSet<>(packageContext.getGlobalAllowedMethods());
-
-            if (allowedMethodsEls.getLength() > 0) {
-                Node n = allowedMethodsEls.item(0).getFirstChild();
-                if (n != null) {
-                    String s = n.getNodeValue().trim();
-                    if (s.length() > 0) {
-                        allowedMethods.addAll(TextParseUtil.commaDelimitedStringToSet(s));
+            // Fix for WW-5029 (concatenate all possible text node children)
+            Node allowedMethodsNode = allowedMethodsEls.item(0);
+            if (allowedMethodsNode != null) {
+                NodeList allowedMethodsChildren = allowedMethodsNode.getChildNodes();
+                for (int i = 0; i < allowedMethodsChildren.getLength(); i++) {
+                    Node allowedMethodsChildNode = allowedMethodsChildren.item(i);
+                    String childNodeValue = (allowedMethodsChildNode != null ? allowedMethodsChildNode.getNodeValue() : "");
+                    childNodeValue = (childNodeValue != null ? childNodeValue.trim() : "");
+                    if (childNodeValue.length() > 0) {
+                        allowedMethods.addAll(TextParseUtil.commaDelimitedStringToSet(childNodeValue));
```

Comment:
If I understand the code correctly, this change will not really fix the problem as for each text node of the `allowed-methods` element, you are independently parsing the comma delimited list of values and adding it to the list of allowed methods.
e.g. if the SAX parser reads the `allowed-methods` tag as `[ "method1,me", "thod2" ]`, this implementation will retrieve the following allowed methods : `[ "method1", "me", "thod2" ]`.

I think you should instead construct a single string with the content of all nodes and call `TextParseUtil.commaDelimitedStringToSet` only once, something like this (not tested):
```
if (allowedMethodsEls.getLength() > 0) {
	Node allowedMethodsNode = allowedMethodsEls.item(0);
	NodeList allowedMethodsChildren = allowedMethodsNode.getChildNodes();
	StringBuilder nodeValue = new StringBuilder();
	for (int i = 0; i < allowedMethodsChildren.getLength(); i++) {
		if (allowedMethodsChildren.item(i).getNodeType() == Node.TEXT_NODE) {
			String val = allowedMethodsChildren.item(i).getNodeValue();
			if (val != null) {
				nodeValue.append(val);
			}
		}
	}
	String val = value.toString().trim();
	if (val.length() > 0) {
		allowedMethods.addAll(TextParseUtil.commaDelimitedStringToSet(val));
	}
}
```

Reply:
Hello @mclem.
Thanks for pointing out that weakness ... looks like I was focusing too much on the other elements and missed that.  :blush:
Wasn't able to come up with an effective unit test for this scenario, and the 1st attempt at a fix still worked when the SAX parser provided a single node of content.
The latest commit provides updated logic similar to what you suggested above, which should work in theory ...